### ISSUE TITLE
"no exported member" errors when using NodeNext module/moduleResolution in TypeScript

### ISSUE THREAD

**Issue by bhouston** [pmndrs/three-stdlib#342]
- `three` version: n/a
- `@types/three` version: n/a
- `three-stdlib` version: 2.29.4

### Problem description:

I ran into this bug and it took a while to figure out. It turns out that this library (as well as threes-stdlib) are not compatible with "NodeNext" module/moduleResolution, e.g. if you tsconfig.json looks like this:

```json
    "module": "NodeNext",
    "moduleResolution": "NodeNext",
```

And you try to use this library like this:

Code:
```ts
import { GLTF, GLTFLoader } from 'three-stdlib'
```

You will get compile errors via TSC like this:
```bash
src/Scene.tsx:17:16 - error TS2305: Module '"three-stdlib"' has no exported member 'GLTFLoader'.
```

You can work around the bug by switching your tsconfig.json to something like this instead:

```json
    "module": "ESNext",
    "moduleResolution": "Node",
```

### Relevant code:

Included in the above description.

### Suggested solution:

I suspect there is some way to fix this?  But I do not yet know.  One clue is that @react-three/fiber and @react-three.drei work properly with these tsconfig.json settings.

**Comment by rosskevin** [comment]
Yes, this is bizarre.  I just switched to `nodenext` because https://github.com/sindresorhus/meow required it for a similar reason as mentioned here https://github.com/sindresorhus/meow/issues/248#issuecomment-1793680332

That links to a related babylon issue https://github.com/BabylonJS/Babylon.js/issues/13797

@bhouston how did you end up working around it?

**Comment by CodyJasonBennett** [comment]
> It looks like it could be spec required file extension.

That is the issue here, in addition to needing the same treatment with `.d.cts` for CommonJS.

**Comment by CodyJasonBennett** [comment]
Had to revert. We're not given the tools necessary to meet these conditions without directly extending the TypeScript compiler. I do plan to deprecate this package as soon as three accepts any upstream contributions here, and they then require ESM for WebGPU (top-level await). Until then, this library effectively serves as a CJS build for `three/addons`.
